src/main/java/ArraySortedBag.java

```diff
@@ -120,62 +110,61 @@ public boolean add(T element) {
         return true;
     }
 
+
     @Override
-    public T removeFirst() {
+    public boolean remove(T element) {
         if (isEmpty()) {
-            throw new NoSuchElementException("Removing first from an empty bag.");
+            throw new NoSuchElementException("Empty bag");
         }
-        T returnElement = bag[0];
-        shiftLeft(0);
+        int index = find(element);
+        if (index == NOT_FOUND){
+            throw new NoSuchElementException(Objects.toString(element));
+        }
+        shiftLeft(index);
         rear--;
-        return returnElement;
+        return true;
     }
 
     @Override
-    public T removeLast() {
+    public T removeFirst() {
         if (isEmpty()) {
-            throw new NoSuchElementException("Removing last from an empty bag.");
+            throw new NoSuchElementException("Empty bag");
         }
-        T returnElement = bag[rear - 1];
-        bag[rear - 1] = null;
+        T returnElement = bag[0];
+        shiftLeft(0);
         rear--;
         return returnElement;
     }
 
     @Override
-    public boolean remove(T element) {
+    public T removeLast() {
         if (isEmpty()) {
-            throw new NoSuchElementException("Removing from an empty bag.");
-        }
-        int removeIndex = sentinelIndexOf(element);
-        if (removeIndex == NOT_FOUND) {
-            throw new NoSuchElementException("Element not contained in bag.");
+            throw new NoSuchElementException("Empty bag");
         }
-        T returnElement = bag[removeIndex];
-        shiftLeft(removeIndex);
+        T returnElement = bag[rear - 1];
         rear--;
-        return true;
+        return returnElement;
     }
 
     @Override
     public T first() {
         if (isEmpty()) {
-            throw new NoSuchElementException("First from an empty bag.");
+            throw new NoSuchElementException("Empty bag");
         }
         return bag[0];
     }
 
     @Override
     public T last() {
         if (isEmpty()) {
-            throw new NoSuchElementException("Last from an empty bag.");
+            throw new NoSuchElementException("Empty bag");
         }
-        return bag[rear - 1];
+        return bag[rear-1];
     }
 
     @Override
     public boolean contains(T target) {
-        return sentinelIndexOf(target) != NOT_FOUND;
+        return find(target) != NOT_FOUND;
     }
 
     @Override
```

Comment:
I suppose this could quit early if it sees a bigger value, but, whatevz.

Reply:
May be worthwhile still. It's just an `else if a < b then break` in `find`.